zds/forum/commons.py

```diff
@@ -0,0 +1,208 @@
+            try:
+                forum_pk = int(request.POST.get('forum'))
+            except (KeyError, ValueError, TypeError):
+                raise Http404
+            forum = get_object_or_404(Forum, pk=forum_pk)
+            topic.forum = forum
+
+            # If the topic is moved in a restricted forum, users that cannot read this topic any more un-follow it.
+            # This avoids unreachable notifications.
+            followers = TopicFollowed.objects.filter(topic=topic)
+            for follower in followers:
+                if not forum.can_read(follower.user):
+                    follower.delete()
+            messages.success(request, _(u"Le sujet {0} a bien été déplacé dans {1}.").format(topic.title, forum.title))
+        else:
+            raise PermissionDenied
+
+    @staticmethod
+    def perform_edit_info(topic, data, editor):
+        (tags, title) = get_tag_by_title(data.get('title'))
+        topic.title = title
+        topic.subtitle = data.get('subtitle')
+        topic.save()
+
+        post = topic.first_post()
+        post.update_content(data.get('text'))
+        post.update = datetime.now()
+        post.editor = editor
+        post.save()
+
+        # add tags
+        topic.tags.clear()
+        topic.add_tags(tags)
+        return topic
+
+
+class PostEditMixin(object):
+    @staticmethod
+    def perform_hide_message(request, post, user, data):
+        if post.author == user or user.has_perm('forum.change_post'):
+            post.alerts.all().delete()
+            post.is_visible = False
+            post.editor = user
+
+            if user.has_perm('forum.change_post'):
+                post.text_hidden = data.get('text_hidden')
+
+            messages.success(request, _(u'Le message est désormais masqué.'))
+        else:
+            raise PermissionDenied
+
+    @staticmethod
+    def perform_show_message(post, user):
+        if user.has_perm('forum.change_post'):
+            post.is_visible = True
+            post.text_hidden = ''
+        else:
+            raise PermissionDenied
+
+    @staticmethod
+    def perform_alert_message(request, post, user, alert_text):
+        alert = Alert()
+        alert.author = user
+        alert.comment = post
+        alert.scope = Alert.FORUM
+        alert.text = alert_text
+        alert.pubdate = datetime.now()
+        alert.save()
+
+        messages.success(request, _(u'Une alerte a été envoyée à l\'équipe concernant ce message.'))
+
+    @staticmethod
+    def perform_useful(post):
+        post.is_useful = not post.is_useful
+        post.save()
+
+    @staticmethod
+    def perform_unread_message(post, user):
+        if TopicFollowed.objects.filter(user=user, topic=post.topic).count() == 0:
+            TopicFollowed(user=user, topic=post.topic).save()
+
+        topic_read = TopicRead.objects.filter(topic=post.topic, user=user).first()
+        if topic_read is None:
+            if post.position > 1:
```

Comment:
plutot que d'avoir une succession de `if` ici, vaudrait mieux mettre `if toto and tata` ?